src/cmds/pbsdsh.c

```diff
@@ -95,6 +107,96 @@ bailout(int sig)
  * @return - Void
  *
  */
+
+
+int
+host_match(char *line, char *host)
+{
+	if (NULL == host ) return 0;
+	int len = strlen(line);
+	static char domain[PBS_MAXHOSTNAME + 1];
+	char fullhost[PBS_MAXHOSTNAME + 1];
+	static struct in_addr addr;
+	static int addrvalid = -1;
+
+	if (line[len - 1] == '\n')
+		line[len - 1] = '\0';
+
+	if (strcmp(line, host) == 0)
+		return 1;
+
+	if (addrvalid == -1) {
+		addr.s_addr = inet_addr(host);
+		addrvalid = (addr.s_addr == INADDR_NONE) ? 0 : 1;
+	}
```

Comment:
Since you pass in host every time, it might be different.  You only do this the first time since addrvalid is static.

Reply:
True, I took these part from pbs_tmrsh which is safe as it only check one host per process. while  it is safe here as -h is not compatible with '-c' or '-n'(you are not suppose have a multi -h, aren't you?), I'd like to chop the static prefix here which leads to confusion.